atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -37,17 +37,29 @@ def __init__(self, tasker, workflow, parent_registry=None, parent_registry_insec
         self.parent_registry = parent_registry
         self.parent_registry_insecure = parent_registry_insecure
 
-    def run(self):
-        """
-        pull base image
-        """
-        base_image = self.workflow.builder.base_image
+    def resolve_base_image(self, build_json):
+        spec = build_json.get("spec")
+        try:
+            image_id = spec['triggeredBy'][0]['imageChangeBuild']['imageID']
+            base_image = ImageName.parse(image_id)
```

Comment:
I think we should move the `ImageName.parse(image_id)` call into an `else` block, to make sure we don't catch exceptions from it. This call should not raise exceptions; if it does it indicates a bug.
```
try:
    image_id = ...  # this might raise TypeError, KeyError, or IndexError
except (TypeError, KeyError, IndexError):
    base_image = ...
else:
    base_image = ImageName.parse(image_id)  # any exceptions will propagate
```

Reply:
fixed.